golang/cosmos/x/swingset/client/cli/tx.go

```diff
@@ -42,10 +42,34 @@ func GetTxCmd(storeKey string) *cobra.Command {
 	return swingsetTxCmd
 }
 
+// parseInput handles an input argument that may refer to a file.
+// If the input does not start with a @, it is returned as-is.
+// The slice after the @ represents the filename. If the filename is -
+// a single line from stdin is returned, otherwise the full content
+// of the file is returned.
+func parseInput(input string) (output string, err error) {
+	if input[0] != '@' {
+		return input, nil
+	}
+
+	fname := input[1:]
+	if fname == "-" {
+		// Reading from stdin.
+		_, err = fmt.Scanln(&output)
+		return output, err
+	} else {
```

Comment:
Nit: when the if-clause ends in a jump (a `return` in this case), you don't need the `else` clause and can save a level of indentation.